### ISSUE TITLE
Problem with Scala 2.13.0-RC1

### ISSUE THREAD

**Issue by gslowikowski** [mockito/mockito-scala#95]
I found very strange problem with Scala `2.13.0-RC1`. Simple mocking code, that works with all previous versions, fails with this version.

Check the example project https://github.com/gslowikowski/mockito-scala-2.13.0-RC1-problem

It fails with the same error with Scalatest `3.0.8-RC2` and `org.scalatestplus.mockito.MockitoSugar` class.

**Comment by ultrasecreth** [comment]
@gslowikowski Interesting, I'll take a look over the weekend

**Comment by ultrasecreth** [comment]
@gslowikowski it seems to be a problem with the compiler not handling final methods correctly, I've reported it [here](https://github.com/scala/bug/issues/11485).

In the meantime, you can use inline-mocking as a workaround.

inline-mocking uses a different stubber that allows to mock final methods and classes, but is still not the default in Mockito as it doesn't support 100% of the use cases and feedback from the community is still being collected.

To enable it, just create a simple text file `test/resources/mockito-extensions/org.mockito.plugins.MockMaker` that just contains the text `mock-maker-inline`.

Let's hope RC2 or the final version includes a fix for this.

**Comment by gslowikowski** [comment]
Thank you @bbonanno

**Comment by ultrasecreth** [comment]
Update: PR that should fix this issue [here](https://github.com/scala/scala/pull/7980)

**Comment by smarter** [comment]
To try this PR, you can set `scalaVersion := 2.13.0-pre-02f59fb-SNAPSHOT` after adding to your build:
```scala
  resolvers += "scala-integration" at
    "https://scala-ci.typesafe.com/artifactory/scala-pr-validation-snapshots/",
```
Would be interested in knowing if everything works with that.

**Comment by ultrasecreth** [comment]
@smarter It was a wise call, it works for the simple example in the PR, but it doesn't in a more complex scenario, I couldn't figure out what type of hierarchy creates it, but if you run 

```scala
java.lang.reflect.Modifier.isFinal(classOf[scala.reflect.io.AbstractFile].getMethod("toIterable").getModifiers)
```

it yields `true`, which causes everything to blow up

**Comment by smarter** [comment]
@bbonanno Once a version of the scala-library compiled with the fixed scala-compiler is published, this should fix itself.

**Comment by ultrasecreth** [comment]
@smarter Fair enough, I'll give it a go as son as RC2 is released

Thanks for your help!

**Comment by smarter** [comment]
The following should work:
```scala
scalaVersion := 2.13.0-pre-cf9468e
```
```scala
 resolvers += "scala-integration" at
    "https://scala-ci.typesafe.com/artifactory/scala-integration/",
```

**Comment by ultrasecreth** [comment]
@smarter it works as expected

Thanks!

**Comment by erikvanoosten** [comment]
2.13.0-RC2 is already out :)

**Comment by ultrasecreth** [comment]
@erikvanoosten Yes!, but I have to wait for my depencencies to be released compiled for RC2 (so far cats-core is the missing one).
I'll release a version supporting RC2 as soon as all my dependencies do :)

**Comment by magnolia-k** [comment]
cats-core 2.0.0-M2 for Scala 2.13.0-RC2 is released!

https://search.maven.org/artifact/org.typelevel/cats-core_2.13.0-RC2/2.0.0-M2/jar

**Comment by ultrasecreth** [comment]
Fixed by #112
